pkg/resources/validation/cluster/provisioningcluster.go

```diff
@@ -135,6 +147,36 @@ func validateCreatorID(request *webhook.Request, oldCluster, cluster *v1.Cluster
 	return nil
 }
 
+func (p *provisioningClusterValidator) validateClusterName(request *webhook.Request, response *webhook.Response, cluster *v1.Cluster) error {
+	if request.Operation != admissionv1.Create {
+		return nil
+	}
+
+	_, err := p.mgmtClusterClient.Get(cluster.Name, metav1.GetOptions{})
```

Comment:
Do we really need to call this `Get` when we're not using its response?
Is this validation really concerned with whether the cluster with such a name exists?

Reply:
We want to know if a v3.Cluster with such a name exists. If it does, then creating the provisioning cluster with this name is valid (the controller is creating it as expected). If no v3.Cluster exists, then the user is trying to create a provisioning cluster with an invalid name and we wish to block it.

Reply:
Thank you for the quick explanation response, I though I was missing some important detail here.